jzp-code/kukdo/project/cloud-storage/cloud-storage-parent/cloud-storage-web/src/main/java/com/xiaomi/controller/FileController.java

```diff
@@ -0,0 +1,68 @@
+package com.xiaomi.controller;
+
+import com.xiaomi.ApiResponse;
+import com.xiaomi.model.FileMetadata;
+import com.xiaomi.service.FileService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+@RequestMapping("/api")
+@RestController
+public class FileController {
+
+    private final FileService fileService;
+
+    @Autowired
+    public FileController(FileService fileService) {
+        this.fileService = fileService;
+    }
+
+    @GetMapping("/files")
+    public List<String> listFiles() {
+        return fileService.getAllFiles();
+    }
+
+    @PostMapping("/files")
+    public ResponseEntity<FileMetadata> uploadFile(@RequestParam("file") MultipartFile file) {
+        FileMetadata metadata = fileService.storeFile(file);
+        return ResponseEntity.ok(metadata);
+    }
+
+    @GetMapping("/files/{Id}")
+    public ResponseEntity<Resource> downloadFile(@PathVariable Long Id) {

```

Comment:
argument name (`Id`) should start with a lowercase letter

Reply:
> argument name (`Id`) should start with a lowercase letter
got it, I will replace all the `Id` with `id` in the later version.